src/features/auth/components/AuthForm/AuthForm.tsx

```diff
@@ -0,0 +1,110 @@
+'use client';
+
+import './AuthForm.scss';
+
+import { Routes } from '@core/values';
+import { Credentials } from '@entities/session';
+import { AuthFormConfig, AuthFormType, RegexpValidation, ValidationMessages } from '@features/auth';
+import { useLoader } from '@features/loader';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { TextField } from '@shared/ui/TextField';
+import clsx from 'clsx';
+import Link from 'next/link';
+import { FC, memo } from 'react';
+import { useForm } from 'react-hook-form';
+import { object, string } from 'yup';
+
+type Props = {
+  type: AuthFormType;
+  onFormSubmit: (data: Credentials, reset: () => void) => void;
+};
+
+const AuthForm: FC<Props> = function ({ type, onFormSubmit }) {
```

Comment:
I'm using function expression to create a context. Since this is a full-fledged abstraction/entity that has a context

I think option 2 satisfies my wishes😂

function Component(props: Props) {
  // component code goes here
}